### ISSUE TITLE
Serialization exception when reading from Remote Hazelcast cluster

### ISSUE THREAD

**Issue by gokhanoner** [hazelcast/hazelcast-jet#1336]
As described in the #957 , Jet requires Predicate & Projections be available in remote cluster. But even if we don't use them in the code, we still suffer.

```
Pipeline.create().drawFrom(Sources. predicate(Constants.MAP_1, clientConfig,START_FROM_OLDEST))
                .drainTo(Sinks.remoteMap(Constants.MAP_2, clientConfig));
```

Above fails if the remote cluster is just a regular Hazelcast cluster with below exception. One can see that it's the Projection & Predicate Jet is adding internally that causes the issue. In the code samples also, all remote source examples uses an embedded Hazelcast node which uses Jet jar & all related classes. It seems only possible solution is to star the remote Hazelcast nodes using Jet jars since `predicate` requires `DistributedPredicate` and `DistributedFunctions` which are Jet only types.

```
Caused by: com.hazelcast.nio.serialization.HazelcastSerializationException: java.lang.ClassNotFoundException: com.hazelcast.jet.impl.connector.StreamEventJournalP$1
```

**Comment by viliam-durina** [comment]
It was fixed in #1306 in 3.0-SNAPSHOT. Can you test it's fixed in your case?

**Comment by gokhanoner** [comment]
@viliam-durina yes, when using Hazelcast Jet 3.0-SNAPSHOT & Hazelcast IMDG 3.12-BETA-2 as remote cluster, it works (Since new helper classes you added on that version only)

Thanks for the quick response.
